### PR TITLE
Improves TileView

### PR DESCRIPTION
**Pull request by tig**
This PR to https://github.com/gui-cs/Terminal.Gui/pull/2258 fixes some of the issues I found during my code review and re-write of `ConfigurationEditor` to use it.

If the direciton of these changes is acceptable to @tznind, I will merge https://github.com/gui-cs/Terminal.Gui/pull/2258 into `develop` and we can continue there.

### CODE REVIEW

File: Terminal.Gui/Views/TileView.cs [hunk 1]
```diff
@@ -111,17 +189,26 @@ protected virtual void OnSplitterMoved (int idx)
...
+			if (splitterLines != null) {
+				foreach (var sl in splitterLines) {
+					sl.Dispose ();
+				}
+			}
 			splitterLines = new List<TileViewLineView> ();
 
 			RemoveAll ();
+			foreach (var tile in tiles) {
+				tile.ContentView.Dispose ();
```
Comment (tznind) [inline Terminal.Gui/Views/TileView.cs:209]: Disposing a `View` disposes the whole tree which could be a problem if the user maintains references to stuff in tiles and wants to put them back in the new configuration.

For example the following use case?

```
// Create TextView

// Add to Tile

// Do some stuff

// Rebuild for Tile count +1

// Add TextView back in at Tile element 0 (user won't know but TextView has actually already been disposed)
```

**Comment by tznind** [comment]
These are great changes thanks, Good catches on the focus/enter and title changing. Only thing I'm nervous about is the disposal of content during the lifetime of `TileView` specifically in `RebuildForTileCount`.

I think we have to balance the needs of the Scenario (everything must be disposed) by thinking of patterns of use where premature disposing might be a problem.

A solution might be removing the content from the `Tile.ContentView` prior to disposing when in `RebuildForTileCount`? I can try and work up a test case that shows what I am worrying about...

File: Terminal.Gui/Views/TileView.cs (hunk 1 above)
Comment (tznind) [reply Terminal.Gui/Views/TileView.cs:209]: Nevermind think I am panicing about nothing.  Just wrote this test and it passes

```csharp
[Fact,AutoInitShutdown]
public void TestDisposal()
{
	var tv = GetTileView (20,10);

	var myReusableView = new DisposeCounter ();

	// I want my view in the first tile
	tv.Tiles.ElementAt (0).ContentView.Add (myReusableView);
	Assert.Equal (0, myReusableView.DisposalCount);

	// I've changed my mind, I want 3 tiles now
	tv.RebuildForTileCount (3);

	// but I still want my view in the first tile
	tv.Tiles.ElementAt (0).ContentView.Add (myReusableView);
	Assert.Multiple (
		()=>Assert.Equal (0, myReusableView.DisposalCount)
		,()=> {
			tv.Dispose ();
			Assert.Equal (1, myReusableView.DisposalCount); 
		});
}
```

Comment (tznind) [reply Terminal.Gui/Views/TileView.cs:209]: Actually `tiles` is always empty at this point because line 198 initializes it to a new list.

**Comment by tznind** [comment]
Ok @tig  I've added the disposal tests to my splitcontainer branch and they are passing there.  I think what we want to do is handle the Dispose in the scenario instead of the `TileView`?

**Comment by tig** [comment]
I renamed View to ContentView because I have an intuition Tile should really inherit from FrameView like WizardPane does. 

In didnt dive deep enough to really figure it out. But I may try to refactor Wizard
to use SplitView to see. It might require enhancements to FrameView, but if we could push more code down to FrameView (which already manages a subview container and deals with Dispose correctly) it might be a win.

**Pull request merged**